storage/src/desktop/storage_reference_desktop.h

```diff
@@ -28,6 +28,25 @@ namespace firebase {
 namespace storage {
 namespace internal {
 
+enum StorageReferenceFn {
+  kStorageReferenceFnDelete = 0,
+  kStorageReferenceFnDeleteInternal,
+  kStorageReferenceFnGetBytes,
+  kStorageReferenceFnGetBytesInternal,
+  kStorageReferenceFnGetFile,
+  kStorageReferenceFnGetFileInternal,
+  kStorageReferenceFnGetDownloadUrl,
```

Comment:
I think GetDownloadUrl could also fail and should get a retry.

Reply:
GetDownloadUrl is currently implemented as just calling GetMetadata and then returning the download url from the metadata, so it effectively gets retry logic for free from the internal call to GetMetadata.